### PR TITLE
[Blueprint] Support removing depends_on in update

### LINKED ISSUE #2594: Azure Blueprint Artifact Update
Hi,

Going through the Azure CLI for using the Blueprint module, I see that you can add a --depends-on to a specific artifact. However, the CLI doesn't seem to support removing this property or setting it to be empty: []. When you try to set the artifact's "dependsOn" property to empty or "", it gives an error when trying to publish the new blueprint. The only way to fix this, that I have found so far, is by exporting the entire Blueprint with artifacts using Powershell and removing the "dependsOn" property manually inside of the artifact file. Then publishing a new version works as expected.

Is there any documentation to direct me in the right direction? I can't seem to find anything, and I would prefer to use the Azure CLI if possible rather than having to use both. Thank you!

V/r

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 47abba92-7835-67eb-8d1d-4c6a812ddd6d
* Version Independent ID: de3d0485-5660-44d7-8c80-7f5843a66fde
* Content: [az blueprint artifact template](https://docs.microsoft.com/en-us/cli/azure/ext/blueprint/blueprint/artifact/template?view=azure-cli-latest#ext_blueprint_az_blueprint_artifact_template_update)
* Content Source: [latest/docs-ref-autogen/ext/blueprint/blueprint/artifact/template.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/latest/docs-ref-autogen/ext/blueprint/blueprint/artifact/template.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve #2594

To remove depends_on relationships between artifacts, you can use `--depends-on`, `--depends-on=` or `--depends-on ""` in artifact update commands.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Blueprint

**Comment by fengzhou-msft** [comment]
@Juliehzl @qiaozha I updated `update` commands with `nargs='*'` so it can support removing the property when updating while keeping `nargs='+'` in `create` commands. This may be a good example of when to use `nargs='*'` and `nargs='+'` in CLI.

File: src/blueprint/azext_blueprint/_params.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def load_arguments(self, _):
         c.argument('artifact_name', help='A unique name of this resource group artifact.')
         c.argument('display_name', help='Display name of this resource group artifact.')
         c.argument('description', help='Description of the blueprint artifact.')
-        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('depends_on', nargs='*', help='Artifacts which need to be deployed before the specified artifact.')
```
Comment (qwordy) [inline src/blueprint/azext_blueprint/_params.py:116]: May be you can add "To remove depends_on relationships between artifacts, you can use --depends_on, --depends_on= or --depends_on "" in artifact update commands." to help message.

**Comment by qwordy** [comment]
> To remove depends_on relationships between artifacts, you can use --depends_on, --depends_on= or --depends_on "" in artifact update commands.

--depends-on is better

**Comment by qiaozha** [comment]
> @Juliehzl @qiaozha I updated `update` commands with `nargs='*'` so it can support removing the property when updating while keeping `nargs='+'` in `create` commands. This may be a good example of when to use `nargs='*'` and `nargs='+'` in CLI.

Actually you can use directive the set that param as required: false. Then it will use nargs=‘*’
```
cli:
    cli-diretive:
        - where: 
            group: groupName
            op: opName
            param: paramName
          set:
            required: false
```

**Review by Juliehzl — APPROVED** [review]
LGTM

File: src/blueprint/azext_blueprint/_params.py (hunk 1 above)
Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/_params.py]: updated

**Review by qwordy — APPROVED** [review]

**Pull request merged**